### PR TITLE
Zach/ftdi wrapper

### PR DESCRIPTION
**Pull request by AtomicCharge**

### CODE REVIEW

File: src/utility/ftdi_serial.hpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+			DTR_DSR = FT_FLOW_DTR_DSR,
+			XON_XOFF = FT_FLOW_XON_XOFF
+		};
+
+	public:
+		FTDISerial();
+		~FTDISerial();
+		bool open(const std::string& serial_number);
+		bool close();
+		bool read(std::string& msg, unsigned int numChars = 0);
```
Comment (irwineffect) [inline src/utility/ftdi_serial.hpp]: Can we change these std::strings to std::vector<uint8_t>?

File: src/utility/ftdi_serial.hpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		void setFlowControl(FTDISerial::flow flow, unsigned char x_on = '\0',
+							unsigned char x_off = '\0');
+
+		static bool populateDeviceList();
+		static const std::vector<std::string>& getDeviceList();
+
+	private:
+		FT_HANDLE m_handle;
+		FT_STATUS m_status;
+		DWORD m_baud_rate = 9600;
```
Comment (irwineffect) [inline src/utility/ftdi_serial.hpp]: Is it possible to use fixed-width datatypes like unint32_t?

File: src/utility/ftdi_serial.cpp [hunk 3]
```diff
@@ -0,0 +1,163 @@
+#include "ftdi_serial.hpp"
+#include <cstring>
+#include <unistd.h>
+
+namespace rs
+{
+	std::vector<std::string> FTDISerial::m_dev_list;
```
Comment (irwineffect) [inline src/utility/ftdi_serial.cpp]: convert all tabs to spaces

**Review by irwineffect — CHANGES_REQUESTED** [review]

File: src/utility/ftdi_serial.cpp (hunk 3 above)
Comment (bkallaher) [inline src/utility/ftdi_serial.cpp]: Fixed in b95aaa3

Comment (irwineffect) [inline src/utility/ftdi_serial.cpp]: looks good here!

**Comment by AtomicCharge** [comment]
Implemented requested changes.

**Review by irwineffect — APPROVED** [review]
Looks good!

**Review by irwineffect — CHANGES_REQUESTED** [review]
I added this to the build and tried to compile, and it won't compile because I don't have ftd2xx.h. We'll want to work together to add that dependency to our robosub debian repo before this gets merged into dev.

**Review by irwineffect — APPROVED** [review]

**Comment by skallaher** [comment]
@bkallaher mentioned adding I2C support. Would that be feasible?

**Pull request merged**